### ISSUE TITLE
Running tensorflow 2 official models with horovod

### ISSUE THREAD

**Issue by kingwales1** [horovod/horovod#1843]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) Tensorflow
2. Framework version: 2.0
3. Horovod version: latest
4. MPI version: 4.0
5. CUDA version: 10.1
6. NCCL version:
7. Python version: 3.7
8. OS and version: Ubuntu 16.04
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
I have successfully set up the distributed environment and run the example with Horovod. And I also know that if I want to run the benchmark on TensorFlow 1 in a distributed setup, e.g. 4 nodes, following the tutorial, the submission should be:

$ horovodrun -np 16 -H server1:4,server2:4,server3:4,server4:4 \
    python scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py \
        --model resnet101 \
        --batch_size 64 \
        --variable_update horovod \
        --data_dir /path/to/imagenet/tfrecords \
        --data_name imagenet \
        --num_batches=2000

But now I want to run the TensorFlow 2 [official models](https://github.com/tensorflow/models), for example BERT model. What command should I use? is it the same as tensorflow 1? Does horovod support the tensorflow 2 official models?

**Comment by abditag2** [comment]
Yes, Horovod supports TF2 and the command is the same. We have several examples in Horovod repo for Tensorflow 2 that you can take a look at.

**Comment by kingwales1** [comment]
Hi @abditag2 Thanks! I did try the horovod example. But I want to know if I can run the official models from tensorflow 2, this [repo](https://github.com/tensorflow/models)?

**Comment by abditag2** [comment]
You can. You need to make a few lines of changes though. 

1. hvd.init()
2. wrap your optimizer with hvd.DistributedOptimizer
3. scale you learning rate


For example, mnist official model is https://github.com/tensorflow/models/blob/master/official/vision/image_classification/mnist_main.py
and its modification with horovod is 
https://github.com/horovod/horovod/blob/master/examples/tensorflow2_mnist.py

**Comment by kingwales1** [comment]
Hi @abditag2 Thanks for your help. I have two more questions.
1, when i ran the mnist example provided, here is the log from one iteration:

[1,1]<stderr>:INFO:tensorflow:loss = 0.12649937, step = 150 (0.487 sec)
[1,1]<stderr>:I0426 17:18:37.099774 140350276830976 basic_session_run_hooks.py:260] loss = 0.12649937, step = 150 (0.487 sec)
[1,3]<stderr>:INFO:tensorflow:loss = 0.027115755, step = 150 (0.489 sec)
[1,3]<stderr>:I0426 17:18:37.059092 140574145767168 basic_session_run_hooks.py:260] loss = 0.027115755, step = 150 (0.489 sec)
[1,2]<stderr>:INFO:tensorflow:loss = 0.061646484, step = 150 (0.487 sec)
[1,2]<stderr>:I0426 17:18:37.059473 140485845481216 basic_session_run_hooks.py:260] loss = 0.061646484, step = 150 (0.487 sec)
[1,0]<stderr>:INFO:tensorflow:loss = 0.01051317, step = 150 (0.488 sec)
[1,0]<stderr>:I0426 17:18:37.101015 140593541785344 basic_session_run_hooks.py:260] loss = 0.01051317, step = 150 (0.488 sec)

Is it the loss value from each machine? How to obtain global loss? Can I configure that?

2, Can I obtain the computation and communication time in each step? For example, in one step, I want to know how much time is spent on computation, and how much on communication.

Many thanks,

**Comment by kingwales1** [comment]
Hi @abditag2 Fabin, Can you help me look at these two problems?
